pkg/odo/genericclioptions/context.go

```diff
@@ -147,6 +211,47 @@ func newContext(command *cobra.Command, createAppIfNeeded bool) *Context {
 			}
 		}
 	}
+	return app
+}
+
+// resolveComponent resolves component
+func resolveComponent(command *cobra.Command, lci *config.LocalConfigInfo, context *Context) string {
+	var cmp string
+	cmpFlag := FlagValueIfSet(command, ComponentFlagName)
+	if len(cmpFlag) == 0 {
+		// retrieve the current component if it exists if we didn't set the component flag
+		cmp = lci.GetName()
+	} else {
+		// if flag is set, check that the specified component exists
+		context.checkComponentExistsOrFail(cmpFlag)
+		cmp = cmpFlag
+	}
+	return cmp
+}
+
+// newContext creates a new context based on the command flags, creating missing app when requested
+func newContext(command *cobra.Command, createAppIfNeeded bool) *Context {
+	client := client(command)
+
+	// Get details from config file
+	configFileName := FlagValueIfSet(command, ContextFlagName)
+	if configFileName != "" {
+		fAbs, err := pkgUtil.GetAbsPath(configFileName)
+		util.LogErrorAndExit(err, "")
+		configFileName = fAbs
+	}
+
+	// Check for valid config
+	lci, err := getValidConfig(command)
+	if err != nil {
+		util.LogErrorAndExit(err, "")
+	}
+
+	// resolve project
+	ns := resolveProject(command, client, lci)
+
+	// resolve application
+	app := resolveApp(command, createAppIfNeeded, lci)
```

Comment:
I don't see the importance of this bcoz it's only going to be component create that will not pass these flagsas far as I can think of but probably defer it to post new workflow implementation for all commands after which we can probably revisit this

Reply:
Yes we can revisit this after. Its more about making things clear and concise. It also makes this function much smaller which is what is favoured, especially in go